test/e2e/sagemaker/tests/test_hpo.py

```diff
@@ -0,0 +1,151 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+"""Integration tests for the SageMaker HyperParameterTuning API.
+"""
+
+import boto3
+import pytest
+import logging
+from typing import Dict
+import time
+
+from sagemaker import SERVICE_NAME, service_marker, CRD_GROUP, CRD_VERSION
+from sagemaker.replacement_values import REPLACEMENT_VALUES
+from common.resources import load_resource_file, random_suffix_name
+from common import k8s
+
+RESOURCE_PLURAL = 'hyperparametertuningjobs'
+
+
+@pytest.fixture(scope="module")
+def sagemaker_client():
+    return boto3.client('sagemaker')
+
+
+@pytest.fixture(scope="module")
+def xgboost_hpojob():
+    resource_name = random_suffix_name("xgboost-hpojob", 32)
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["HPO_JOB_NAME"] = resource_name
+
+    hpojob = load_resource_file(
+        SERVICE_NAME, "xgboost_hpojob", additional_replacements=replacements)
+    logging.debug(hpojob)
+
+    # Create the k8s resource
+    reference = k8s.CustomResourceReference(
+        CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL, resource_name, namespace="default")
+    resource = k8s.create_custom_resource(reference, hpojob)
+    resource = k8s.wait_resource_consumed_by_controller(reference)
+
+    assert resource is not None
+
+    yield (reference, resource)    
+
+    # Delete the k8s resource if not already deleted by tests
+    try:
+        k8s.delete_custom_resource(reference)
+    except:
+        pass
+
+
+@service_marker
+@pytest.mark.canary
+class TestHPO:
+    def _get_created_hpo_job_status_list(self):
+        return ["InProgress", "Completed"]
```

Comment:
https://github.com/aws-controllers-k8s/community/pull/706#discussion_r589906565

Reply:
we can't move this to "stoped_hpo_list" because then we won't be able to verify if delete is working.

Reply:
A job can go from InProgress -> Stopping -> Completed/Stopped depending on when it was stopped. So its fine to expect Completed if job is stopped.

If completed is part of created state then we don't know if its because it has deleted or job actually completed. Either we need a longer job here or we need to poll more frequently